src/Layout/Presentation/SingleDistrict/SingleDistrict.tsx

```diff
@@ -36,11 +38,12 @@ export class SingleDistrict extends React.Component<SingleDistrictProps, {}> {
     };
 
     render() {
+        const calculateVulnerable = isQuotientAlgorithm(this.props.algorithm);
         const currentDistrictResult = this.getDistrictResult(this.props.districtSelected);
-        const vulnerableMap = getVotesToVulnerableSeatMap(currentDistrictResult!);
-        const quotientMap = getQuotientsToVulnerableSeatMap(currentDistrictResult!);
-        const vulnerable = getVulnerableSeatByQuotient(currentDistrictResult!);
-        const vulnerableVotes = getVulnerableSeatByVotes(currentDistrictResult!);
+        const vulnerableMap = calculateVulnerable ? getVotesToVulnerableSeatMap(currentDistrictResult!) : undefined;
+        const quotientMap = calculateVulnerable ? getQuotientsToVulnerableSeatMap(currentDistrictResult!) : undefined;
+        const vulnerable = calculateVulnerable ? getVulnerableSeatByQuotient(currentDistrictResult!) : undefined;
+        const vulnerableVotes = calculateVulnerable ? getVulnerableSeatByVotes(currentDistrictResult!) : undefined;
```

Comment:
Again, weird undefined returns.

Reply:
See my earlier comment for an answer to this.

Reply:
The function itself should probably return undefined, and we could encapsulate all these changes in an if block.

Reply:
What do you mean by encapsulating all the changes in an if block?

Reply:
Given that there are 4 lines using the same conditional (which should probably be renamed to shouldCalculateVulnerable) we might consider having two returns (one for each case) and rewriting a bit of the code to use components, but I don't think this is important right now. We can create clean up issues when/if we have time for something as small as this.